test/images/resource-consumer/README.md

```diff
@@ -0,0 +1,83 @@
+# Resource Consumer
+
+## Overview
+
+Resource Consumer is a tool which allows to generate cpu/memory utilization in a container.
+The reason why it was created is testing kubernetes autoscaling.
+Resource Consumer can help with autoscaling tests for:
+- cluster size autoscaling,
+- horizontal autoscaling of pod - changing the size of replication controller,
+- vertical autoscaling of pod - changing its resource limits.
+
+## Usage
+
+Resource Consumer starts an HTTP server and handle sended requests.
+It listens on port given as a flag (default 8080).
+Action of consuming resources is send to the container by a POST http request.
+Each http request creates new process.
+Http request handler is in file resource_consumer_handler.go 
+
+The container consumes specified amount of resources:
+
+- CPU in millicores,
+- Memory in megabytes.
+
+
+
+###Consume CPU http request
+
+- suffix "ConsumeCPU",
+- parameters "millicores" and "durationSec".
+
+Consumes specified amount of millicores for durationSec seconds.
+Consume CPU uses "./consume-cpu/consume-cpu" binary (file consume-cpu/consume_cpu.go).
+When CPU consumption is too low this binary uses cpu by calculating math.sqrt(0) 10^7 times
+and if consumption is too high binary sleeps for 10 millisecond.
+One replica of Resource Consumer cannot consume more that 1 cpu.
+
+###Consume Memory http request
+
+- suffix "ConsumeMem",
+- parameters "megabytes" and "durationSec".
+
+Consumes specified amount of megabytes for durationSec seconds.
+Consume Memory uses stress tool (stress -m 1 --vm-bytes megabytes --vm-hang 0 -t durationSec).
+
+###CURL example
+```console
+$ kubectl run resource-consumer --image=gcr.io/google_containers/resource_consumer:beta
+$ kubectl expose rc resource-consumer --create-external-load-balancer=true --port=8080 --target-port=8080
+$ kubectl get services resource-consumer
+```
+
+There are two IPs first one is internal, second one is the external load-balanced IP. Both serving port 8080. (Use second one)
+
+```console
+$ curl --data "millicores=300&durationSec=600" http://104.197.103.250:8080/ConsumeCPU.
+```
+
+300 millicores will be consumed for 600 seconds.
```

Comment:
I would clarify that 300 millicores will be consumed for 600 seconds on the pod that received the curl request.

In addition, maybe discuss what happens if you make overlapping requests back-to-back on the same pod.